site/content/docs/user/nvidia-dranet.md

```diff
@@ -0,0 +1,460 @@
+```
+
+It works on the single pod. Now let's try between the two pods.
+
+```sh
+root@nccl-gib-test-0:/usr/bin# /usr/local/gib/scripts/run_nccl_tests.sh   -t all_gather -b 1K -e 8G   nccl-gib-test-0 10.68.5.39 -g 2
+Initializing SSH...
+Hello from nccl-gib-test-0
+Warning: Permanently added '[10.68.5.39]:222' (ECDSA) to the list of known hosts.
+Hello from 10.68.5.39
+Generating hostfiles for 2 hosts:
+nccl-gib-test-0
+10.68.5.39
+# nThread 1 nGpus 1 minBytes 1024 maxBytes 8589934592 step: 2(factor) warmup iters: 50 iters: 100 agg iters: 1 validation: 1 graph: 0
+#
+# Using devices
+#  Rank  0 Group  0 Pid  25060 on nccl-gib-test-0 device  0 [0000:cb:00] NVIDIA B200
+#  Rank  1 Group  0 Pid  25055 on nccl-gib-test-0 device  1 [0000:cc:00] NVIDIA B200
+#  Rank  2 Group  0 Pid   2078 on nccl-gib-test-1 device  0 [0000:97:00] NVIDIA B200
+#  Rank  3 Group  0 Pid   2055 on nccl-gib-test-1 device  1 [0000:c4:00] NVIDIA B200
+NCCL version 2.26.6+cuda12.8
+#
+#                                                              out-of-place                       in-place
+#       size         count      type   redop    root     time   algbw   busbw #wrong     time   algbw   busbw #wrong
+#        (B)    (elements)                               (us)  (GB/s)  (GB/s)            (us)  (GB/s)  (GB/s)
+```
+
+Uh oh! We can gather the info between the pods but we can't run data? Running `ip a` shows us the issue.
+
+```sh
+root@nccl-gib-test-0:/diagnostic# ip a
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+2: eth0@if44: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1460 qdisc noqueue state UP group default qlen 1000
+    link/ether b6:76:b9:08:9c:e5 brd ff:ff:ff:ff:ff:ff link-netnsid 0
+    inet 10.68.3.40/24 brd 10.68.3.255 scope global eth0
+       valid_lft forever preferred_lft forever
+```
+
+There are no NICs to run the data! This is where DRANet can fix things!
+
+#### Nvidia DRA + DRANet
+
+We create one more `ResourceClaimTemplate`, for the RDMA devices on the node, along with a `DeviceClass` for the RDMA device.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: DeviceClass
+metadata:
+  name: dranet
+spec:
+  selectors:
+    - cel:
+        expression: device.driver == "dra.net"
+```
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaimTemplate
+metadata:
+  name: 2-nic
+spec:
+  spec:
+    devices:
+      requests:
+      - name: nic
+        deviceClassName: dranet
+        count: 2
+        selectors:
+        - cel:
+            expression: device.attributes["dra.net"].rdma == true &&
+              (
+                (device.attributes["dra.net"].ifName.startsWith("gpu") &&
+                 device.attributes["dra.net"].ifName.endsWith("rdma0") &&
+                 int(device.attributes["dra.net"].ifName.substring(3, 4)) < 2)
```

Comment:
we need to explain that this achieves GPU and NIC alignment because google machines use this convention to create the interfaces with the name of the closest GPU to the nic `gpuXrdma0`